scorex-consensus/src/main/scala/scorex/consensus/nxt/NxtLikeConsensusModule.scala

```diff
@@ -95,27 +97,53 @@ class NxtLikeConsensusModule(AvgDelay: Long = 5.seconds.toMillis)
     } else Future(None)
   }
 
+  private def effectiveBalance[TT](account: Account)(implicit transactionModule: TransactionModule[TT]) =
+    transactionModule.blockStorage.state.asInstanceOf[BalanceSheet].balanceWithConfirmations(account.address, EffectiveBalanceDepth)
+
   private def calcGeneratorSignature(lastBlockData: NxtLikeConsensusBlockData, generator: PublicKeyAccount) =
     hash(lastBlockData.generationSignature ++ generator.publicKey)
 
   private def calcHit(lastBlockData: NxtLikeConsensusBlockData, generator: PublicKeyAccount): BigInt =
-    BigInt(1, calcGeneratorSignature(lastBlockData, generator).take(8))
-
-  private def calcBaseTarget(lastBlockData: NxtLikeConsensusBlockData,
-                             lastBlockTimestamp: Long,
-                             currentTime: Long): Long = {
-    val eta = currentTime - lastBlockTimestamp
-    val prevBt = BigInt(lastBlockData.baseTarget)
-    val t0 = bounded(prevBt * eta / AvgDelay, prevBt / 2, prevBt * 2)
-    bounded(t0, 1, Long.MaxValue).toLong
+    BigInt(1, calcGeneratorSignature(lastBlockData, generator).take(8).reverse)
+
+  /**
+    * BaseTarget calculation algorithm fixing the blocktimes.
+    *
+    */
+  private def calcBaseTarget[TT](prevBlock: Block, timestamp: Long) (implicit transactionModule: TransactionModule[TT]): Long = {
+    val height = transactionModule.blockStorage.history.heightOf(prevBlock).get
+    val prevBaseTarget = consensusBlockData(prevBlock).baseTarget
+    if (height % 2 == 0) {
+      val lastBlocks: Seq[Block] = transactionModule.blockStorage.history.lastBlocks(3)
+      val block = lastBlocks.head
+      val blocktimeAverage = (timestamp - block.timestampField.value) / lastBlocks.size
+      val bt =
```

Comment:
Type `if` on the same line as val, like `val bt = if (blocktimeAverage > AvgDelayInSeconds) {`